mqtt/mqttd/src/broker/bootstrap/edgehub.rs

```diff
@@ -152,23 +134,69 @@ where
         broker_ready.send(()).expect("ready signal");
     });
 
-    let device_id = env::var(DEVICE_ID_ENV).context(DEVICE_ID_ENV)?;
-
-    // Start the sidecars
-    let system_address = config.listener().system().addr().to_string();
-    let (sidecar_shutdown_handle, sidecar_join_handle) =
-        start_sidecars(broker_handle, system_address, device_id).await?;
-
     // Start serving new connections
     let state = server.serve(shutdown).await?;
 
-    // Shutdown the sidecars
-    sidecar_shutdown_handle.shutdown()?;
-    sidecar_join_handle.await??;
-
     Ok(state)
 }
 
+#[derive(Debug, Error)]
+pub enum SidecarError {
+    #[error("Failed to shutdown command handler")]
+    CommandHandlerShutdown(#[from] CommandHandlerError),
+}
+
+#[derive(Clone, Debug)]
+pub struct SidecarShutdownHandle {
+    command_handler_shutdown: ShutdownHandle,
+}
+
+impl SidecarShutdownHandle {
+    pub fn new(command_handler_shutdown: ShutdownHandle) -> Self {
+        Self {
+            command_handler_shutdown,
+        }
+    }
+
+    pub async fn shutdown(self) -> Result<(), SidecarError> {
+        self.command_handler_shutdown
+            .shutdown()
+            .await
+            .map_err(SidecarError::CommandHandlerShutdown)
+    }
```

Comment:
Forgot to mention, since bridge shutdown is not implemented, the bridge will never finish once signaled to shutdown. This results in the broker process being killed. Once bridge shutdown works then broker process can stop properly.

If we don't want this bad behavior in the short term, we can simply not return the bridge join handle from the below ```start_sidecars()``` method.

Reply:
This happens only if we have an upstream bridge enabled